src/pages/ReportDetailsPage.tsx

```diff
@@ -260,12 +260,25 @@ function ReportDetailsPage({policies, report, route, reportMetadata}: ReportDeta
     }, [report, personalDetails, shouldOpenRoomMembersPage]);
     const connectedIntegration = getConnectedIntegration(policy);
 
+    const caseID = useMemo((): CaseID => {
+        // 3. MoneyReportHeader
+        if (isMoneyRequestReport || isInvoiceReport) {
+            return CASES.MONEY_REPORT;
+        }
+        // 2. MoneyRequestHeader
+        if (isSingleTransactionView) {
+            return CASES.MONEY_REQUEST;
+        }
+        // 1. HeaderView
+        return CASES.DEFAULT;
+    }, [isInvoiceReport, isMoneyRequestReport, isSingleTransactionView]);
```

Comment:
Can we remove `useMemo` and just use `if else` here? Memoization shouldn't be necessary.